librz/util/iterator.c

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2024 billow <billow.fun@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_util/rz_iterator.h>
+#include <rz_util/rz_assert.h>
+
+RZ_API RzIterator *rz_iterator_new(F_RzIterator next, F_RzIterator_FREE free, F_RzIterator_FREE free_u, void *u) {
```

Comment:
doxygen

Reply:
Also, can those function pointers be NULL? Please add corresponding mark/checks if necessary.